packages/story-editor/src/components/videoTrim/useVideoTrimMode.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useFeature } from 'flagged';
+import { useCallback, useMemo } from '@web-stories-wp/react';
+
+/**
+ * Internal dependencies
+ */
+import { useCanvas, useStory } from '../../app';
+
+function useVideoTrimMode() {
+  const isVideoTrimEnabled = useFeature('enableVideoTrim');
+  const { isEditing, isTrimMode, setEditingElementWithState, clearEditing } =
+    useCanvas(
+      ({
+        state: { isEditing, editingElementState: { isTrimMode } = {} },
+        actions: { setEditingElementWithState, clearEditing },
+      }) => ({
+        isEditing,
+        isTrimMode,
+        setEditingElementWithState,
+        clearEditing,
+      })
+    );
+  const { selectedElement } = useStory(({ state: { selectedElements } }) => ({
+    selectedElement: selectedElements.length === 1 ? selectedElements[0] : null,
+  }));
+  const { isTranscoding, isMuting, isTrimming, local } =
+    selectedElement.resource;
+
+  const toggleTrimMode = useCallback(() => {
+    if (isEditing) {
+      clearEditing();
+    } else {
+      setEditingElementWithState(selectedElement.id, {
+        isTrimMode: true,
+      });
+    }
+  }, [isEditing, clearEditing, setEditingElementWithState, selectedElement]);
+
+  const hasTrimMode = useMemo(
+    () =>
+      selectedElement?.type === 'video' &&
+      isVideoTrimEnabled &&
+      !isTranscoding &&
+      !isMuting &&
+      !isTrimming &&
+      local,
```

Comment:
Local means local to file system, not to the editor. Once the editor upload the file, is not local anymore

Reply:
Should there also be a check for if the video is from coverr?

Reply:
Oh, wait, we also need to check that it's not a hotlinked video from an external source, no? Or can you also transcode those - and do they then become WP assets? Are we "allowed" to do that?

Reply:
@miina, is there a way to recognize that a video is hotlinked from an external source?

Reply:
Ah yes, Coverr videos too, I guess.

Reply:
Filed this as #8982 for follow-up.